addons/mail/static/src/public/discuss_public_boot.js

```diff
@@ -57,38 +70,26 @@ Component.env = legacyEnv;
     ]);
     mapLegacyEnvToWowlEnv(Component.env, env);
     odoo.isReady = true;
-    legacyServiceRegistry.add('messaging', MessagingService.extend({
-        messagingValues: {
-            autofetchPartnerImStatus: false,
-        },
-    }));
-    await mount(MainComponentsContainer, document.body, { env, templates, dev: env.debug });
-    createAndMountDiscussPublicView(templates);
-})();
 
-async function createAndMountDiscussPublicView(templates) {
-    const messaging = await Component.env.services.messaging.get();
-    // needed by the attachment viewer
-    const DialogManager = getMessagingComponent('DialogManager');
-    await mount(DialogManager, document.body, {
-        templates,
-        env: Component.env,
-        dev: Component.env.isDebug(),
-    });
+    const messaging = await env.services.messaging.get();
     messaging.models['Thread'].insert(messaging.models['Thread'].convertData(data.channelData));
     const discussPublicView = messaging.models['DiscussPublicView'].create(data.discussPublicViewData);
     if (discussPublicView.shouldDisplayWelcomeViewInitially) {
         discussPublicView.switchToWelcomeView();
     } else {
         discussPublicView.switchToThreadView();
     }
-    const DiscussPublicView = getMessagingComponent('DiscussPublicView');
-    await mount(DiscussPublicView, document.body, {
-        templates,
-        env: Component.env,
-        dev: Component.env.isDebug(),
+
+    registry.category('main_components').add('DialogManager', {
+        Component: getMessagingComponent('DialogManager'),
+    });
+    registry.category('main_components').add('DiscussPublicView', {
+        Component: getMessagingComponent('DiscussPublicView'),
         props: {
             record: discussPublicView,
+            className: 'position-absolute top-0 w-100',
```

Comment:
What is that extra stuff, and why not in template?

Reply:
Added to the template. The `MainComponentContainer` doesn't specify any height (and I don't think it should do so). This is an issue because the `DiscussPublicView` has the `h-100` class that can't work if it's parent is not either specifying height or a flex container. Those classes allow the `DiscussPublicView` to take the whole screen.